src/v/coproc/wasm_event.h

```diff
@@ -46,12 +49,25 @@ std::variant<wasm::errc, event_action> get_event_action(const model::record&);
 /// \brief errc::none if the record is valid AND the checksum passes validation
 wasm::errc verify_event_checksum(const model::record&);
 
-/// \brief performs the strictest form of validation, 'none' if all checks pass
-wasm::errc validate_event(const model::record&);
+/// \brief Structure for parsed event from model::record. It contains raw data
+/// of the event and header with action and coproc_type
+struct parsed_event {
```

Comment:
I would change that name. It suggest that `parsed` is transient. We store it in a map which suggest a state. I find it a little bit confusing.